src/core/tests/Test_ZypperPackageManager.py

```diff
@@ -183,5 +184,51 @@ def test_install_package_failure(self):
         # test for unsuccessfully installing a package
         self.assertEquals(package_manager.install_update_and_dependencies('selinux-policy.noarch', '3.13.1-102.el7_3.16', simulate=True), Constants.FAILED)
 
+    def test_get_process_tree_from_package_manager_msg_success(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+
+        package_manager = self.container.get('package_manager')
+        self.assertIsNotNone(package_manager)
+
+        # Get this process' pid to test output with a real running process
+        test_process_pid = os.getpid()
+        self.assertIsNotNone(test_process_pid)
+
+        # Create example package manager message and include this process' pid
+        package_manager_msg = 'Output from package manager: | System management is locked by the application with pid {} (/usr/bin/zypper).'
+        package_manager_msg = package_manager_msg.format(str(test_process_pid))
+
+        # Test to make sure a valid string was returned with process information
+        self.assertIsNotNone(package_manager.get_process_tree_from_package_manager_msg(package_manager_msg))
+
+    def test_get_process_tree_from_package_manager_msg_failure_nonexistent_process(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+
+        package_manager = self.container.get('package_manager')
+        self.assertIsNotNone(package_manager)
+
+        # Test with a nonexistent pid
+        test_process_pid = -1
```

Comment:
Similar to the comment above, you need to mock the response on running get_process_tree_command